libcxx/include/locale

```diff
@@ -546,8 +548,8 @@ int __num_get<_CharT>::__stage2_float_loop(
     }
     return 0;
   }
-  ptrdiff_t __f = std::find(__atoms, __atoms + 32, __ct) - __atoms;
-  if (__f >= 32)
```

Comment:
Why is this changed from 32 to 28. Please add the reason to the commit message too.